src/test/java/app/sami/languageWeb/testUtils/factories/LanguageGradesFactory.java

```diff
@@ -0,0 +1,22 @@
+package app.sami.languageWeb.testUtils.factories;
+
+import app.sami.languageWeb.language.models.Language;
+import app.sami.languageWeb.language.models.LanguageGrades;
+import app.sami.languageWeb.testUtils.Randomize;
+import app.sami.languageWeb.user.models.User;
+import app.sami.languageWeb.user.repos.UserRepository;
+
+import java.util.UUID;
+
+public class LanguageGradesFactory {
+
+    public static LanguageGrades generateFromUsers(UUID reviewedId, UUID reviewerId, double grade, Language language){
```

Comment:
you should keep the generation from factories fully random or partially random and then edit it:

```
var grades = LanguageGradesFactory.generate().withUserId(userId).withLanguage(ENGLISH); // fully random, then specify english
var evalGrades = grades.withGrade(10).withEmitterId(evalId);
var userGrades = grades.withGrade(50).withEmiiterId(userId);
```

The `.with` methods come from lombok `@With` annotation and they always return a new object keeping the original one unmodified.